### PR TITLE
Correct irrelevant inheritance in Mutect2

### LINKED ISSUE #5352: Unused arguments in documentation for Mutect2
### Tool(s) or class(es) involved
Mutect2 and FilterMutectCalls

### Description 
Because both `M2ArgumentCollection` and `M2FiltersArgumentCollection` extend `AssemblyBasedCallerArgumentCollection`, both `Mutect2` and `FilterMutectCalls` display all assembly and caller arguments in the documentation/help even if those arguments don't actually do anything. For example both tools have the argument `--contamination-fraction-to-filter` which has the description:

```
Fraction of contamination in sequencing data (for all samples) to aggressively remove.

If this fraction is greater is than zero, the caller will aggressively attempt to remove contamination 
through biased down-sampling of reads. Basically, it will ignore the contamination fraction of reads for 
each alternate allele. So if the pileup contains N total bases, then we will try to remove 
(N * contamination fraction) bases for each alternate allele.
```

This argument definitely doesn't do anything in `FilteMutectCalls` but I also don't think it's hooked up to do anything in `Mutect2` either (at least when I tried giving it a high value I still got the same calls). This is by design because Mutect has other ways of handling contamination, but the argument is still displayed in both tools' documentation which is confusing. There are other arguments that have the same issue where it's unclear if they do anything in Mutect or not.

### PR DESCRIPTION
**Pull request by davidbenjamin**
Closes #5352.  This removes unused command line arguments from Mutect2.

@LeeTL1220 This PR looks big but it's really just a lot of moving around `static` methods and giving `HaplotypeCallerArgumentCollection` a `StandardCallerCollection` as a member instead of having`AssemblyBasedCallerCollection` inherit from it.

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerArgumentCollection.java [hunk 1]
```diff
@@ -133,7 +103,7 @@ public ReadThreadingAssembler createReadThreadingAssembler() {
 
     @Advanced
     @Argument(fullName = SMITH_WATERMAN_LONG_NAME, doc = "Which Smith-Waterman implementation to use, generally FASTEST_AVAILABLE is the right choice", optional = true)
-    public SmithWatermanAligner.Implementation smithWatermanImplementation = SmithWatermanAligner.Implementation.JAVA;
+    public SmithWatermanAligner.Implementation smithWatermanImplementation = SmithWatermanAligner.Implementation.FASTEST_AVAILABLE;
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerArgumentCollection.java]: @davidbenjamin Did you intend for this PR to make substantive changes to default values for arguments? You present it as a pure refactoring branch, but it looks like there are changes like this embedded in it. We deliberately have not yet turned the native SmithWaterman on by default due to our current inability to continuously test it.

Comment (davidbenjamin) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerArgumentCollection.java]: @droazen That's an accident -- I was just curious about its effect on the M2 integration test runtime and forgot to revert.  I intend everything to be refactoring.  This does include simplifying the assembly argument collections, which is not strictly necessary for the purposes of this PR.  I could strip out things like that if you would prefer.

Comment (droazen) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerArgumentCollection.java]: Ah ok, thanks for the clarification! Can you do a pass to check that additional changes like this didn't slip in (ie., changes to argument defaults)?

Comment (davidbenjamin) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerArgumentCollection.java]: Absolutely!

Comment (davidbenjamin) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerArgumentCollection.java]: . . . That was the only one, fortunately.

**Pull request merged**